hongit-api/src/main/kotlin/com/devfloor/hongit/api/article/application/ArticleService.kt

```diff
@@ -66,18 +71,56 @@ class ArticleService(
     }
 
     @Transactional(readOnly = true)
-    fun showAllByBoardId(boardId: Long): List<ArticleFeedResponse> {
+    fun showAllByBoardId(
+        boardId: Long,
+        page: Int,
+        pageSize: Int,
+        sort: ArticleSortType?,
+        options: List<Long>?
+    ): List<ArticleFeedResponse> {
         val board = boardRepository.findByIdOrNull(boardId)
             ?: EntityNotFoundException.notExistsId(Board::class, boardId)
 
-        return articleRepository.findAllByBoard(board)
-            .map {
-                ArticleFeedResponse(
-                    article = it,
-                    articleOptions = articleOptionRepository.findAllByArticle(it),
-                    articleFavorites = articleFavoriteRepository.findAllByArticle(it),
-                )
+        return when (sort) {
+            ArticleSortType.CREATED -> {
+                val pageRequest: PageRequest = PageRequest.of(page, pageSize, Sort.by("createdAt"))
+                articleRepository.findAllByBoard(board, pageRequest).content
+                    .map {
+                        ArticleFeedResponse(
+                            article = it,
+                            articleOptions = articleOptionRepository.findAllByArticle(it),
+                            articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                            page = page,
+                            totalArticleCount = articleRepository.countAllByBoard(board)
+                        )
+                    }
+            }
+            ArticleSortType.VIEW_COUNT -> {
+                val pageRequest: PageRequest = PageRequest.of(page, pageSize)
+                articleRepositoryCustom.findAllByBoardSortByViewCount(board, pageRequest).map {
+                    ArticleFeedResponse(
+                        article = it,
+                        articleOptions = articleOptionRepository.findAllByArticle(it),
+                        articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                        page = page,
+                        totalArticleCount = articleRepository.countAllByBoard(board)
+                    )
+                }
```

Comment:
```suggestion
                articleRepositoryCustom.findAllByBoardSortByViewCount(board, pageRequest)
                    .map {
                        ArticleFeedResponse(
                            article = it,
                            articleOptions = articleOptionRepository.findAllByArticle(it),
                            articleFavorites = articleFavoriteRepository.findAllByArticle(it),
                            page = page,
                            totalArticleCount = articleRepository.countAllByBoard(board)
                        )
                    }
```

map을 한 라인 내리면 코드가 좀 더 보기 좋은 것 같아서 이건 추천으로 달아둘게

Reply:
반영완료